salt/modules/virt.py

```diff
@@ -2918,16 +2920,23 @@ def _get_with_unit(node):
                 )
             if mem:
                 if isinstance(mem, dict):
-                    mem = str(_handle_unit(mem.get("current", 0)))
+                    # setMemoryFlags takes memory amount in KiB
+                    new_mem = (
+                        int(_handle_unit(mem.get("current")) / 1024)
+                        if "current" in mem.keys()
```

Comment:
`if "current" in mem` is not sufficient?

Reply:
right, should be, I'll fix it